adapters/richaudience/richaudience.go

```diff
@@ -200,3 +199,15 @@ func parseImpExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpRichaudience, error) {
 
 	return &richaudienceExt, nil
 }
+
+func getMediaType(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
```

Comment:
This implies that, for a multiformat request that includes video, you will always bid a video ad or not bid at all. You may want to consider passing an explicit media type to give your bidder more flexibility in bidding. Also in OpenRTB 2.6, there will be an `mType` parameter to explicitly provide this.

Reply:
In an openrtb2 request, you can have a multiformat impression. For example the `banner` object and the `video` object may both be populated in an imp. This means that you could bid with an ad that is a banner, or an ad that is a video. The bid has to be one or the other. It is an issue with openrtb 2.5 that this ambiguity in the bid exists, prebid server is trying to resolve this issue by asking bidders to be explicit. Openrtb 2.6 will fix this by including an explicit `mType` field for advertisers to declare what sort of bid they have supplied. But for now, we are working with what we have.

This function you have here assumes that in a banner/video multiformat impression, you will bid a video or not at all. If this is true, you are limiting your options on bidding. If this is false, and you might bid a banner when you have a high value banner ad to deliver, then you are compounding the ambiguity problem by reporting video when you are actually bidding with a banner ad. In the first case, this code is fine, but you are limiting yourself. In the second case, we ask that you find a way to be more accurate in your bid type reporting.

Reply:
Thanks for your explain, I think understood.
I have resolved this in next commit, thanks for your comments

Reply:
This is worse. Now it will only check the last imp (It loops through all the imps, but the last imp will overwrite the other results) and applies that media type to all imps. Please at least revert to the previous code.

My point above has nothing to do with multiple imps. A single imp that defines both banner and video gives you the option of bidding with either a banner ad or a video ad. Your code assumes if the video option is given, you will always bid with video and not consider a banner ad instead. This either limits your options, or results you in reporting the wrong media type. Knowing the correct video type may be important in properly displaying an ad, which is why Prebid Server reports this field. So flagging this incorrectly could result in some of your ads not being displayed properly.

Reply:
I have discarded the changes, because I think best to leave a default bid. I have compared with others adapters and I see same configuration. ¿It's correct?
Kind Regards

Reply:
Yes. My point is that there may be a better way, not that what you had is wrong.